scripts/20210421_has-allowed-email-domains/cleanup.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable */
+
+// Optional: Retrieve forms with fields that has a populated allowedEmailDomains but has no allowed email domains.
+db.getCollection('forms').aggregate([
+  { $unwind: '$form_fields' },
+  {
+    $match: {
+      'form_fields.fieldType': 'email',
+      'form_fields.hasAllowedEmailDomains': false,
+      'form_fields.allowedEmailDomains': { $ne: [] },
+    },
+  },
+  {
+    $group: {
+      _id: '$_id',
+      form_fields: { $addToSet: '$form_fields' },
+    },
+  }
+])
+
+// Retrieve counts of various form states
+// Count forms with email fields isVerifiable=true, hasAllowedEmailDomains=true, non-empty allowedEmailDomains
+// Basically forms with restricted domain email fields
+
+// COUNT_A =
+
+db.getCollection('forms').count({
+  'form_fields.fieldType': 'email',
+  'form_fields.isVerifiable': true,
+  'form_fields.hasAllowedEmailDomains': true,
+  'form_fields.allowedEmailDomains': { $ne: [] },
+})
+
+// Count forms with isVerifiable=true, but hasAllowedEmailDomains=false
+// Basically forms with verification but not restrictions
+
+// COUNT_B = 
+
+db.getCollection('forms').count({
+  'form_fields.fieldType': 'email',
+  'form_fields.isVerifiable': true,
+  'form_fields.hasAllowedEmailDomains': false,
+})
+
+// Count forms with non-empty allowed email domains even if hasAllowedEmailDomains is false, 
+// so we can reset array before removing hasAllowedEmailDomains key.
+
+// COUNT_C = 
+
+db.getCollection('forms').count({
+  'form_fields.fieldType': 'email',
+  'form_fields.hasAllowedEmailDomains': false,
+  'form_fields.allowedEmailDomains': { $ne: [] },
+})
+
+// Count forms with inconsistent email states
+// hasAllowedEmailDomains = true but allowedEmailDomains is empty
+// Should not matter since allowedEmailDomains is the source of truth
+
+// COUNT_D = 
+
+db.getCollection('forms').count({
+  'form_fields.fieldType': 'email',
+  'form_fields.hasAllowedEmailDomains': true,
+  'form_fields.allowedEmailDomains': { $eq: [] },
+})
+
+
+// !!! Main update !!!
+// Set all current fields with hasAllowedEmailDomains = false to empty allowedEmailDomains array [] and remove hasAllowedEmailDomains key.
+// updateCount should be === COUNT_C
+// updateCount = 
+
+db.getCollection('forms').updateMany(
+  {
+    'form_fields.fieldType': 'email',
+    'form_fields.hasAllowedEmailDomains': false,
+    'form_fields.allowedEmailDomains': { $ne: [] },
+  },
+  {
+    $set: {
+      'form_fields.$[elem].allowedEmailDomains': [],
+    },
+    $unset: {
+      'form_fields.$[elem].hasAllowedEmailDomains': 1,
+    }
+  },
+  {
+    arrayFilters: [
+      {
+        'elem.fieldType': 'email',
+        'elem.hasAllowedEmailDomains': false,
+        'elem.allowedEmailDomains': { $ne: [] },
```

Comment:
this removes the need for the `$elemMatch` in the query portion on L76-78, but I think it might still be clearer to add it so that we only match fields which meet all the criteria